lib/rspec/core/configuration.rb

```diff
@@ -202,10 +202,33 @@ def only_failures_but_not_configured?
         only_failures? && !example_status_persistence_file_path
       end
 
-      # @macro add_setting
+      # @macro define_reader
       # If specified, indicates the number of failures required before cleaning
-      # up and exit (default: `nil`).
-      add_setting :fail_fast
+      # up and exit (default: `nil`). Can also be `true` to fail and exit on first
+      # failure
+      define_reader :fail_fast
+
+      # @see fail_fast
+      def fail_fast=(value)
+        case value
+        when true, 'true'
+          @fail_fast = true
+        when false, 'false', 0
+          @fail_fast = false
+        when nil
+          @fail_fast = nil
+        else
+          @fail_fast = value.to_i
+
+          if value.to_i == 0
+            # TODO: in RSpec 4, consider raising an error here.
+            RSpec.warning "Cannot set `RSpec.configuration.fail_fast`" \
+              " to `#{value.inspect}`. Only `true`, `false`, `nil` and integers" \
+              " are valid values."
```

Comment:
But I don't understand. If you provide a 0, it is an integer so it should valid and you should not receive the warning?

Do I miss something? Or should validate only positive integers?

Reply:
`0` is the same as turning it off

Reply:
Yes but if you turn it off with 0 it is still a valid integer?

This not a blocker for me to validate this PR 🙂

Reply:
🤔 Indeed. We could be more specific by saying

> only `true`, `false`, `nil` and positive integers are valid values

I'm not sure if we should be that specific. Keep in mind that, with this PR, we treat `'true'` (string) the same as `true`, yet, it's not mentioned in the warning message above. (and I don't think it should be explicitly mentioned)

Reply:
😄 I think it is good enough